xtraplatform-crs/src/test/groovy/de/ii/xtraplatform/crs/infra/GeoToolsCrsTransformerSpec.groovy

```diff
@@ -0,0 +1,80 @@
+package de.ii.xtraplatform.crs.infra
+
+import de.ii.xtraplatform.crs.domain.CoordinateTuple
+import de.ii.xtraplatform.crs.domain.EpsgCrs
+import spock.lang.Ignore
+import spock.lang.Shared
+import spock.lang.Specification
+
+class GeoToolsCrsTransformerSpec extends Specification{
+
+    @Shared
+    GeoToolsCrsTransformerFactory transformerFactory
+
+    def setupSpec() {
+        transformerFactory = new GeoToolsCrsTransformerFactory()
+    }
+
+    @Ignore
+    def 'CRS transformer test: source and/or target CRS are null'() {
+        when:
+        def transformer = transformerFactory.getTransformer(sourceCrs, targetCrs)
+
+        then:
+        !transformer.isPresent()
+
+        where:
+        src          | trgt         | sourceCrs         | targetCrs
+        "EPSG:5555"  | "null"       | EpsgCrs.of(5555)  | null
+        "null"       | "EPSG:5555"  | null              | EpsgCrs.of(5555)
+        "null"       | "null"       | null              | null
+
+    }
+
+
+    def 'CRS transformer test 3D'() {
+        given:
+            EpsgCrs sourceCrs = EpsgCrs.of(5555)
+            EpsgCrs targetCrs = EpsgCrs.of(4979)
+            double[] source = [420735.071, 5392914.343, 131.96]
+
+        when:
+            GeoToolsCrsTransformer gct = (GeoToolsCrsTransformer) transformerFactory.getTransformer(sourceCrs, targetCrs).get()
+            double[] target = gct.transform3d(source, 1, false)
+
+        then:
+            target == source
```

Comment:
I don't think target and source should be equal.

Reply:
Agreed, this is a bad check, but the test passes. Should I try a different CRS combination?

Reply:
Actually source and target are equal because `transform3d` reuses the input array as output array for performance reasons. 
That is why the test succeeds, but it doesn't make lots of sense. So I replaced source in the assertion with the actual expected transformed coordinates.